src/store/notificationsSlice.ts

```diff
@@ -1,18 +1,22 @@
+import type { ReactElement } from 'react'
 import { createSlice, type PayloadAction } from '@reduxjs/toolkit'
 import type { AlertColor } from '@mui/material'
 import type { AppThunk, RootState } from '@/store'
 import type { LinkProps } from 'next/link'
 
 export type Notification = {
   id: string
-  message: string
+  message: string | ReactElement
   detailedMessage?: string
   groupKey: string
   variant: AlertColor
   timestamp: number
   isDismissed?: boolean
   isRead?: boolean
   link?: { href: LinkProps['href']; title: string }
+  autoHide?: boolean
+  isBanner?: boolean
+  onClose?: () => void
```

Comment:
Did you try using the `Toast` component directly instead of modifying the notification system?

Reply:
Will check if its easy to implement with the fixed positioning, then we can keep the notification system more pure.